### PR TITLE
Cnft fix and daily balances surrogate fix

### PR DESCRIPTION
**Pull request by andrewhong5297**
I think we have a decoded table issue where some of these buy instructions are decoding at 18 decimals instead of 9. But here is a temporary fix.

also adding a fix for the null in unique key, using surrogate function. Will see if this fixes the issue with duplicates on incremental.

# Thank you for contributing to Spellbook!
Please refer to the top of the `readme` in the root of Spellbook to learn how to contribute to Spellbook on DuneSQL.

### CODE REVIEW

**Comment by andrewhong5297** [comment]
I used the surrogate_key generate function @aalan3 mentioned. It looks like it passed for 30 days.

Last time it also passed the incremental check, but then because of nulls in "token_mint_address" then we ended up with duplicates on merge. Hopefully the surrogate key fixes that this time, but would love a look from @aalan3 on this.

File: models/solana_utils/solana_utils_daily_balances.sql [hunk 1]
```diff
@@ -45,6 +45,7 @@ SELECT
       , token_mint_address
       , token_balance
       , token_balance_owner
+      , {{ dbt_utils.generate_surrogate_key(['address', 'token_mint_address', 'day', 1]) }} as unique_address_key
```
Comment (aalan3) [inline models/solana_utils/solana_utils_daily_balances.sql:48]: the `1` can be dropped here.

Comment (andrewhong5297) [reply models/solana_utils/solana_utils_daily_balances.sql]: okay cool! everything else looks good?

**Review by aalan3 — APPROVED** [review]

**Review by jeff-dude — APPROVED** [review]

**Comment by jeff-dude** [comment]
i'm going to cancel CI and merge now

**Pull request merged**